### PR TITLE
Add diesel to the benchmark suite

### PR DESCRIPTION
**Pull request by weiznich**
This commit adds diesel to the continuosly run benchmark suite. Diesel
heavily relies internally on macro generated code. Additionally there
are lots of complicated trait releations used as part of their API.
Therefore this benchmark will be quite sensitive to:
* Performance related changes in the macro expanding code
* Performance related changes while resolving trait bounds

CC #7950

cc @lnicola

### CODE REVIEW

**Comment by lnicola** [comment]
changelog internal add `diesel` to the benchmark suite.

**Comment by lnicola** [comment]
Looks like this should take about 5-6 minutes on CI. It might be pretty painful, but let's tentatively merge it.

I've checked `ripgrep` and `webrender` and they don't seem to have changed in the meanwhile.

bors r+

File: xtask/src/metrics.rs [hunk 1]
```diff
@@ -26,7 +26,7 @@ impl flags::Metrics {
         }
         {
             let _d = pushd("./target/rustc-perf")?;
-            cmd!("git reset --hard 1d9288b0da7febf2599917da1b57dc241a1af033").run()?;
+            cmd!("git reset --hard c52ee623e231e7690a93be88d943016968c1036b").run()?;
```
Comment (matklad) [inline xtask/src/metrics.rs:29]: We should add this hash to JSON we publish to the metrics repo, so that it's easy to see when we update the test suite.

**Comment by lnicola** [comment]
Great, this broke the metrics dashboard :-).

**Pull request merged**